packages/analytics-types/src/plugin.ts

```diff
@@ -27,7 +27,7 @@ export interface DestinationPlugin {
   type: PluginType.DESTINATION;
   setup(config: Config): Promise<undefined>;
   execute(context: Event): Promise<Result>;
-  flush(): Promise<void>;
+  flush(includeQueueBuffer?: boolean): Promise<void>;
```

Comment:
Maybe create an AmplitudeDestinationPlugin which extends destinationPlugin but adds `flush` ? Does browser need `flush` function?

Reply:
Browser does not need flush. All unsent events are persisted in storage provider, and are sent on next init. Does python have flush in plugin interface?

Reply:
Looks like Python has flush interface https://github.com/amplitude/Amplitude-Python/blob/16e56f4174079f3c2a56dd5b695d5def7e2934f1/src/amplitude/plugin.py#L124-L169

Reply:
Hmm this is the implementation right? I think the Plugin interface does not require it. As far as I can remember, only setup() and execute() are required.

Reply:
Basically it is pretty same as the python's implementation. https://github.com/amplitude/Amplitude-Python/blob/16e56f4174079f3c2a56dd5b695d5def7e2934f1/src/amplitude/timeline.py#L43-L48

Timeline has a `flush` method and it loops all destination plugins. And then it calls `flush` function. If destination plugin does not have `flush` interface, the flush from the timeline will be failed.

Reply:
cc @bohan-amplitude 

Then I'd assume it's still not a required interface if we wrapped in try/catch? I wonder if we added that for convenience rather than enforcement. The reason for the pushback here is around friction on developer experience. By adding this line in question, we require developers to add a `flush()` method in their custom destination plugin for both client and server SDKs. I personally prefer to keep it as minimal as possible so developers can quickly learn how to create new plugins. I don't mind having a `flush()` in implementation in our plugin implementation but I would prefer that it is not required in the interface.

Reply:
Move `flush` to a separate interface so it is not required in the destination interface.